Tidy formatting of Amazon S3 checkboxes

rnicoll opened the pull request:
Wrap configuration checkboxes in a list to ensure they do not bump against each other when displayed.

rnicoll commented:
@Xenopathic Not 100% sure how serious you were about this, but here's one possible solution.

Xenopathic commented:
@rnicoll I was serious - it's an ugly UI problem that needed fixing. Are you sure some simple CSS rules won't do the job? Ideally markdown shouldn't be used for styling, but for layout...

rnicoll commented:
I'd sort of presumed if it was a very simple bit of CSS, someone would have done it before now. Haven't got a dev setup for this here, it'll have to wait until I'm in the office tomorrow, but will give it a go with just putting margin on both sides of the checkbox.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/cacf61e1-18db-43f1-b55d-53c2824ced50) was created.

rnicoll commented:
Once more, with feeling

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/cacf61e1-18db-43f1-b55d-53c2824ced50) completed: **No new issues**

Xenopathic commented:
Hmm, slight issue here. The `.added` class only applies for newly added configuration entries, not those that existed when the page was generated (aka saved storages). For that reason, it's probably best not to mess with the CSS on the `.added` class, instead perhaps add a margin to any `label` elements in the configuration? Those contain the checkboxes themselves.

![img](https://cloud.githubusercontent.com/assets/2016878/5782084/5018b896-9db0-11e4-91b6-60c2afd0c4f1.png)

rnicoll commented:
Sorry, missed those were under ".added" (and could have sworn it worked on reload, too). So, question, shouldn't the margin be consistent even if the row hasn't just been added? I'd be tempted to change the selectors instead to remove ".added"?

Again, this is all on a server I don't have access to at home, so will be Monday now.

Xenopathic commented:
Well, it should be, but that `.added` CSS is a bit of a hack to get around a bug we found: see 934229d1fb839190f64f56f27e4849a46725a7a6 :see_no_evil: :laughing:

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/fe5caa15-7501-4747-9827-40cec69ec9c0) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/acfdeab2-d451-4562-a604-1deb919c61f1) was created.

rnicoll commented:
![owncloud](https://cloud.githubusercontent.com/assets/3513077/5783549/dadf1d16-9dbb-11e4-8205-b6ea67da3fa2.png)

Like that?

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/2184a29c-be6e-47ec-b334-4760bc969bbb) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/fe5caa15-7501-4747-9827-40cec69ec9c0) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/acfdeab2-d451-4562-a604-1deb919c61f1) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/2184a29c-be6e-47ec-b334-4760bc969bbb) completed: **No new issues**

Xenopathic commented:
I see the check boxes always going to the left, displacing the 'Region' box:

![img](https://cloud.githubusercontent.com/assets/2016878/5784176/c4356448-9dc1-11e4-8daa-4b3358cb2aec.png)

However, disabling `float: left` and `width: 144px` in the Firefox dev tools makes it look quite good!

![img](https://cloud.githubusercontent.com/assets/2016878/5784201/e1a8b1a6-9dc1-11e4-8468-68117a29bb23.png)

If you remove those properties, leaving just the new margin, I'll be satisfied :smile:

rnicoll commented:
Ah, and I thought I was being so clever getting it to align properly!

Xenopathic commented:
@rnicoll In that case, try the `display: inline-block` property (along with your current `width: 144px` property) - that gives the nice aligned look without the floating issues

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/a45b8c77-e2cc-467c-9630-cc9200e08376) was created.

rnicoll commented:
Moved to inline-block display, seems to work well.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/a45b8c77-e2cc-467c-9630-cc9200e08376) completed: **No new issues**

Xenopathic commented:
:+1: @jancborchardt It'd be great to have this for 8.0 - what do you think?

jancborchardt commented:
Looks good from first view! @rnicoll @Xenopathic can you show a before-after screenshot?

Xenopathic commented:
@jancborchardt Here you go:

Before:
![old](https://cloud.githubusercontent.com/assets/2016878/5937998/5f10903e-a6ee-11e4-8919-5526e5fd0441.png)

After:
![new](https://cloud.githubusercontent.com/assets/2016878/5938001/65689184-a6ee-11e4-90e1-be38f2c01aec.png)

As mentioned above, can this get into oC 8.0? It's a minor CSS change, no risk of breaking anything.

jancborchardt commented:
Nice! :+1: for way better layout. :)

jancborchardt commented:
@rnicoll CSS fixes like that are always appreciated. You’re very welcome in our IRC channel #owncloud-designers as well. :)

karlitschek commented:
nice :+1:

MorrisJobke commented:
Works :+1: Also tested with IE8, IE9, Chromium and Firefox

@DeepDiver1975 It's your decision to merge this.

DeepDiver1975 commented:
@jancborchardt forced me with a :gun: to merge this ...

jancborchardt commented:
DID NOT

LIES

:rooster:

Xenopathic commented:
:see_no_evil:

MorrisJobke commented:
@jancborchardt :hand:
